include/mqt-core/dd/Package.hpp

```diff
@@ -1165,6 +1165,71 @@ template <class Config> class Package {
     return '1';
   }
 
+  mEdge buildMeasOp(const Qubit index, const char measureFor = '0') {
+    mEdge measOp = {};
+    mEdge f = {};
+
+    const auto identityMatrix =
+        std::array{mEdge::one(), mEdge::zero(), mEdge::zero(), mEdge::one()};
+
+    const auto measureMatrix = measureFor == '0'
+                                   ? std::array{mEdge::one(), mEdge::zero(),
+                                                mEdge::zero(), mEdge::zero()}
+                                   : std::array{mEdge::zero(), mEdge::zero(),
+                                                mEdge::zero(), mEdge::one()};
+
+    if (index == 0) {
+      measOp = makeDDNode(static_cast<dd::Qubit>(0), measureMatrix);
+    } else {
+      measOp = makeDDNode(static_cast<dd::Qubit>(0), identityMatrix);
+    }
+
+    for (dd::Qubit p = 1; p < static_cast<dd::Qubit>(qubits()); p++) {
+      if (p == index) {
+        f = makeDDNode(static_cast<dd::Qubit>(0), measureMatrix);
+      } else {
+        f = makeDDNode(static_cast<dd::Qubit>(0), identityMatrix);
+      }
+      measOp = kronecker(f, measOp);
+    }
+    return measOp;
+  }
+
+  std::pair<dEdge, char> measureOneCollapsing(dEdge& e, const Qubit index,
+                                              std::mt19937_64& mt) {
+    char measuredResult = '0';
+
+    auto operation = buildMeasOp(index, '0');
+
+    auto tmp0 = conjugateTranspose(operation);
+    auto tmp1 = multiply(e, densityFromMatrixEdge(tmp0), 0, false);
+    auto tmp2 = multiply(densityFromMatrixEdge(operation), tmp1, 0, true);
+    auto densityMatrixTrace = trace(tmp2);
+
+    std::uniform_real_distribution<fp> dist(0., 1.);
+    if (const auto threshold = dist(mt); threshold > densityMatrixTrace.r) {
+      operation = buildMeasOp(index, '1');
+      tmp0 = conjugateTranspose(operation);
+      tmp1 = multiply(e, densityFromMatrixEdge(tmp0), 0, false);
+      tmp2 = multiply(densityFromMatrixEdge(operation), tmp1, 0, true);
+      measuredResult = '1';
+      densityMatrixTrace = trace(tmp2);
+    }
+
+    incRef(tmp2);
+    dEdge::alignDensityEdge(e);
+    decRef(e);
+    e = tmp2;
+    dEdge::setDensityMatrixTrue(e);
+
+    // Normalize density matrix
+    auto result = e.w / densityMatrixTrace;
+    cn.decRef(e.w);
+    e.w = cn.lookup(result);
+    cn.incRef(e.w);
+    return {e, measuredResult};
+  }
```

Comment:
This method has a slightly different interface then the regular method for vectors.
AFAICT, this need not be the case and you could just be returning the measurement result. `e` is anyway captures by reference and modified as part of the method.

Reply:
You are correct, the interface could be changed to be identical to the regular measure method. However, I don't make use of the variables assumeProbabilityNormalization and epsilon. While I could mark these variables unused, it feels wrong to introduce unused variables simply for similarity. 

Are you sure, you want me to make this change?

Reply:
The `assumeProbabilityNormalization` parameter should just be removed IMO. It is _always_ true. But this does not happen to need here.
What I actually meant here is to "just" change the return type of the function here to `char`.

Reply:
Oh, I see. I changed the interface as suggested.